build_projects/shared-build-targets-utils/Utils/HostVersion.cs

```diff
@@ -107,6 +107,8 @@ public override string ToString()
         //
         public bool IsLocked = true; // Set this variable to toggle muxer locking.
         public VerInfo LockedHostFxrVersion => IsLocked ? new VerInfo(1, 0, 1, "", "", "", CommitCountString) : LatestHostFxrVersion;
+        //We add a different LockedHostFxrVersion for the MSI as the package needs to deviate for issue #1574. This we can update or remove when we have a binary change for the hostfxr.dll (from 1.0.1)
```

Comment:
I would suggest adding a bool here (e.g. fExplicitHostFXRMSIVersion) and set it to true. And then, based upon whether the bool is true or not, either set it to explicit version or the version of LockedHostFXRVersion (this will allow us to simply use this bool to make a switch in future without needing to fix the locations you are changing right now).